src/components/ColorPicker.tsx

```diff
@@ -42,14 +43,40 @@ const ColorSwatch: FC<{
   const dispatch = useDispatch()
   const colors = useSelector(themeColors)
   const fontSize = useSelector(state => state.fontSize)
-  const selected =
-    (color && cursorStyle?.color === color) || (backgroundColor && cursorStyle?.backgroundColor === backgroundColor)
   size = size || fontSize * 1.2
 
+  /** A function that adds an alpha channel to a hex color. */
+  const addAlphaToHex = (hex: string) => {
+    if (hex.length === 7) return hex + 'ff'
+    return hex
+  }
+  const selected = useMemo(() => {
+    const textHexColor = color ? addAlphaToHex(rgbToHex(color)) : undefined
+    const backHexColor = backgroundColor ? addAlphaToHex(rgbToHex(backgroundColor)) : undefined
+    const selectedTextHexColor = cursorStyle?.color ? addAlphaToHex(rgbToHex(cursorStyle?.color)) : undefined
+    const selectedBackHexColor = cursorStyle?.backgroundColor
+      ? addAlphaToHex(rgbToHex(cursorStyle?.backgroundColor))
+      : undefined
+    return !!(
+      (textHexColor && textHexColor === selectedTextHexColor) ||
+      (backHexColor && backHexColor === selectedBackHexColor)
+    )
+  }, [cursorStyle, backgroundColor, color])
   /** Toggles the text color to the clicked swatch. */
   const toggleTextColor = (e: React.MouseEvent | React.TouchEvent) => {
     // stop toolbar button dip
-    e.stopPropagation()
+    // e.stopPropagation()
```

Comment:
# 4. Allowing the click event to propagate breaks the dip behavior

The dip is the small animation that occurs when a toolbar button is pressed.

## Steps to Reproduce

1. Set the cursor on a thought.
2. Set the text color.

## Current Behavior

af1f1757e1

The text color toolbar button remains dipped.

<img width="319" alt="Brave Browser 2024-09-28 08 38 36" src="https://github.com/user-attachments/assets/145b155f-972f-43de-accf-fb1955c9cb46">


## Expected Behavior

`main`

The animation should complete and the button should not dip down.

<img width="318" alt="Brave Browser 2024-09-28 08 38 11" src="https://github.com/user-attachments/assets/913b1619-5c6a-4cb2-8b5c-d093bd3eb4f8">